src/Faker/Extension/GeneratorAwareExtensionTrait.php

```diff
@@ -19,7 +19,7 @@ trait GeneratorAwareExtensionTrait
     /**
      * @return static
      */
-    public function withGenerator(Generator $generator): self
+    public function withGenerator(Generator $generator): Extension
```

Comment:
This change was needed. PHP complained if I was using the trait. 

The optimal solution would be to write `static` here, but that is only supported by PHP8.